### PR TITLE
[fix] (openpi): support Transformers 5.x Gemma APIs

### LINKED ISSUE #438: Transformers version required by Qwen3.5 conflicts with the version expected by OpenPI Gemma
Qwen3.5 requires `transformers >= 5.2.0`, but `starVLA/model/modules/vlm/openpi_transformers/gemma/modeling_gemma.py` imports the old API `check_model_inputs` from `transformers.utils.generic`, which is unavailable in newer Transformers versions.

### PR DESCRIPTION
**Pull request by HaningZS**
## Description

Make the vendored OpenPI Gemma implementation compatible with both the repository-pinned Transformers 4.57 API and Transformers 5.x environments required by Qwen3.5.

## Motivation

Closes #438

On the current `starVLA_dev` baseline, importing the module succeeds with Transformers 4.57.0 but fails with 5.2.0 because `check_model_inputs` was removed. Exercising the real OpenPI action-expert path exposed additional 5.x protocol changes in default RoPE initialization, tied-weight metadata, output capture, and causal-mask arguments.

## Changes

- Prefer the official `merge_with_config_defaults` + `capture_outputs` composition whenever available, falling back to the legacy decorator only on older Transformers.
- Provide the default RoPE initializer expected by 5.x while preserving the 4.57 registry path.
- Use tied-weight metadata accepted by both APIs.
- Detect whether `create_causal_mask` accepts `cache_position`, covering the 4.57 name, 5.2 rename, 5.7 keyword-only cache argument, and current Transformers `main` removal.
- Add production-path tiny forward tests for both PI0 (`use_adarms=False`) and PI0.5 (`use_adarms=True`) action experts using `inputs_embeds` and a 4D mask.
- Cover hidden-state/attention capture, tuple returns, and a main-style causal-mask callable without `cache_position`.

## Testing

- [x] Regression tests pass with Transformers 4.57.0
- [x] Regression tests pass with Transformers 5.2.0
- [x] Regression tests pass with Transformers 5.3.0
- [x] Regression tests pass with Transformers 5.7.0
- [x] Main-style `create_causal_mask` signature without `cache_position` is covered directly
- [x] 11 focused CPU regression tests pass, including both compatibility tests
- [x] A800 CUDA forward passes on GPU 4 with PyTorch 2.5.1+cu121 and Transformers 4.57.6
- [x] `black --check` passes on both changed files
- [x] Ruff passes on the new test; the vendored file retains exactly the same 8 historical Ruff findings as the upstream baseline

The A800 host cannot reach public PyPI and its configured mirror does not provide Transformers 5.2, so the 5.2/5.3/5.7 matrix was run in isolated local CPU environments. The CUDA run separately validates the same PI0/PI0.5 tensor path and backward compatibility on the repository 4.57 line. No checkpoint or external model download is required.

## Breaking Changes

None. The default `requirements.txt` pin remains unchanged.

## Type of Change

- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] New framework / benchmark integration
- [ ] Breaking change
- [ ] Documentation only
- [ ] Refactor (no behavior change)

## Checklist

- [x] No unrelated changes mixed in
- [x] No secrets, API keys, or large binaries committed
- [x] No shared dataloader, trainer, or config files modified
- [x] Regression tests included

### CODE REVIEW

File: starVLA/model/modules/vlm/openpi_transformers/gemma/modeling_gemma.py [hunk 1]
```diff
@@ -412,11 +434,14 @@ def forward(
...
         causal_mask = create_causal_mask(
-            config=self.config,
-            input_embeds=inputs_embeds,
-            attention_mask=attention_mask,
-            cache_position=cache_position,
+            self.config,
+            inputs_embeds,
+            attention_mask,
+            cache_position,
             past_key_values=past_key_values,
```
[the remark this answers is not in the document]
Comment (HaningZS) [reply starVLA/model/modules/vlm/openpi_transformers/gemma/modeling_gemma.py]: Thanks for catching this. I confirmed that Transformers 4.57 through 5.7 still expose `cache_position`, while the official Transformers `main` signature now removes it. Commit `34d1601` detects the installed callable signature once and only passes `cache_position` when supported. I also added a regression test with a main-style `create_causal_mask` callable that has no `cache_position`; the 4.57/5.2/5.3/5.7 matrix still passes.

File: starVLA/model/modules/vlm/openpi_transformers/gemma/modeling_gemma.py [hunk 2]
```diff
@@ -39,7 +39,17 @@
 from transformers.processing_utils import Unpack
 from transformers.utils import TransformersKwargs, auto_docstring, can_return_tuple
 from transformers.utils.deprecation import deprecate_kwarg
-from transformers.utils.generic import check_model_inputs
+
+try:
+    from transformers.utils.generic import check_model_inputs
```
[the remark this answers is not in the document]
Comment (HaningZS) [reply starVLA/model/modules/vlm/openpi_transformers/gemma/modeling_gemma.py]: Thanks—this is valid for Transformers 5.7, which exposes both the deprecated shim and the new decorators. Commit `34d1601` now prefers `merge_with_config_defaults` + `capture_outputs` whenever they are available and falls back to `check_model_inputs` only on older releases. The regression test now asserts `output_hidden_states`, `output_attentions`, and `return_dict=False` behavior across Transformers 4.57, 5.2, 5.3, and 5.7.

**Comment by HaningZS** [comment]
@codex review
